hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -494,23 +562,25 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
 
     // Get the available CG datamaps and prune further.
     DataMapExprWrapper cgDataMapExprWrapper = chooser.chooseCGDataMap(filter.getResolver());
+
     if (cgDataMapExprWrapper != null) {
       // Prune segments from already pruned blocklets
-      pruneSegments(segmentIds, prunedBlocklets);
+      DataMapUtil.pruneSegments(segmentIds, prunedBlocklets);
       List<ExtendedBlocklet> cgPrunedBlocklets;
       // Again prune with CG datamap.
       if (distributedCG && dataMapJob != null) {
-        cgPrunedBlocklets = DataMapUtil.executeDataMapJob(carbonTable, filter.getResolver(),
-            segmentIds, cgDataMapExprWrapper, dataMapJob, partitionsToPrune);
+        cgPrunedBlocklets = DataMapUtil
+            .executeDataMapJob(carbonTable, filter.getResolver(), dataMapJob, partitionsToPrune,
+                segmentIds, invalidSegments, DataMapLevel.CG, new ArrayList<String>());
       } else {
         cgPrunedBlocklets = cgDataMapExprWrapper.prune(segmentIds, partitionsToPrune);
       }
       // since index datamap prune in segment scope,
       // the result need to intersect with previous pruned result
       prunedBlocklets = intersectFilteredBlocklets(carbonTable, prunedBlocklets, cgPrunedBlocklets);
-      ExplainCollector.recordCGDataMapPruning(
-          DataMapWrapperSimpleInfo.fromDataMapWrapper(cgDataMapExprWrapper),
-          prunedBlocklets.size(), getBlockCount(prunedBlocklets));
+      ExplainCollector
```

Comment:
Can u add a check ExplainCollector.enable() then only record the Pruning info

Reply:
done